apps/sms/views/conversation/js/conversation.js

```diff
@@ -547,34 +562,39 @@ var ConversationView = {
   },
 
   beforeEnterThread: function conv_beforeEnterThread(args) {
-    // TODO should we implement hooks to Navigation so that Threads could
-    // get an event whenever the panel changes?
-    Threads.currentId = args.id;
+    var threadId = +args.id;
+    return MessageManager.ensureThreadRegistered(threadId).then(() => {
```

Comment:
1. `ensureThreadRegistered` is clearly view-oriented and (I think) a temporary workaround. In my DB experiment (https://github.com/mozilla-b2g/gaia/pull/28681/files) all Threads getters return promises (and that's why your patch https://github.com/mozilla-b2g/gaia/pull/30798 is useful). And with NGA the logic would be the same: we can get the thread information using a promise in beforeEnter and then just use the cached information.